src/gui/uithememanager.cpp

```diff
@@ -58,6 +61,115 @@ namespace
 
         return {};
     }
+
+    class QRCThemeSource : public UIThemeSource
```

Comment:
I believe that adding couple of conditionals in existing code could be enough to add support for folder based themes instead of implementing these two classes which for the most part duplicate the same logic...

Reply:
IDK but someone has said "Program to interfaces, not implementation"

Reply:
>IDK but someone has said "Program to interfaces, not implementation"

This is actually a good thing and it makes the code easier to maintain and extensible. Maybe you just need some things in their proper places (for example, move logging to a suitable place, as I mentioned earlier).